esp/services/esdl_svc_engine/esdl_binding.cpp

```diff
@@ -860,6 +860,29 @@ void EsdlServiceImpl::handleFinalRequest(IEspContext &context,
 
             soapmsg.append(req.str());
             soapmsg.append("</").append(tgtQueryName).append("></soap:Body>");
+
+            auto cfgGateways = tgtcfg->queryBranch("Gateways");
+            if (cfgGateways)
+            {
+                auto baseXpath = cfgGateways->queryProp("@transformTarget");
+                if (!isEmptyString(baseXpath))
+                {
+                    Owned<IPTree> gws = createPTree("gateways", 0);
+                    VStringBuffer tmpSoap("%s</soap:Envelope>", soapmsg.str());
```

Comment:
I don't think you need a temporary "tmpSoap" variable.  Seems like you can do all the work in soapmsg, since you always swapWith anyway.  Just add the Envelope, transform, toXML, and then remove the Envelope all with soapmsg, no?  I may be missing something.